### PR TITLE
STM32F1 merge other (BluePill and Maple Mini)

### PR DESCRIPTION
**Pull request by fpistm**
Second part of the Arduino_Core_STM32F1 merge. #20 --> now covered by #213 except for USB part

Include above variant from #45

- BLUEPILL-F103C8
- MAPLEMINI-F103CB

Need some review around USB (based on contributor feedback)
~Need review upload menu (avoid maple one for BP)~ Covered by #213
...

### CODE REVIEW

**Comment by RickKimball** [comment]
So I'm looking at this new branch and it doesn't seem like any of the USB issues I mentioned in #45 have been addressed. Is this something that is being looked at?

**Comment by RickKimball** [comment]
BTW: The default upload method of "Maple bootloader" is probably a bad pick for the Other boards. Most people are using a bluepill according to the latest Poll on stm32duino. Even those who are using a maple mini typically update the bootloader to the latest one because it is smaller in flash size and works better in general.

**Comment by fpistm** [comment]
@RickKimball, it's ongoing, they are lot of work/review around USB and this need some time. As you can see, I've rebased regularly this PR.
About the bootloader, you talk about the fact BP has the Maple bootloader in the menu ?
It is plaaned to split Other menu to be able to have correct upoad method and also be able to have some customization as asked in #59

File: boards.txt [hunk 1]
```diff
@@ -310,6 +310,80 @@ Disco.menu.upload_method.STLinkMethod=STLink
...
+Other.menu.upload_method.DFUUploadMethod.upload.tool=maple_upload
+Other.menu.upload_method.DFUUploadMethod.build.upload_flags=-DSERIAL_USB -DVECT_TAB_OFFSET=0x2000
+Other.menu.upload_method.DFUUploadMethod.build.ldscript=bootloader.ld
+Other.menu.upload_method.DFUUploadMethod.upload.usbID=1EAF:0003
+Other.menu.upload_method.DFUUploadMethod.upload.altID=2
+
+Other.menu.upload_method.STLinkMethod=STLink
+Other.menu.upload_method.STLinkMethod.upload.protocol=STLink
+Other.menu.upload_method.STLinkMethod.upload.tool=stlink_upload
+Other.menu.upload_method.STLinkMethod.build.upload_flags=-DSERIAL_USB
```
Comment (fprwi6labs) [inline boards.txt]: Move STLink method before DFUUploadMethod. STLink will be the default Upload method.

File: variants/MAPLEMINI_F103CB/variant.h [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+//In case SPI CS channel is not used we define a default one
+#define BOARD_SPI_OWN_SS        SPI_CHANNELS_NUM
+
+#define SS                      BOARD_SPI_DEFAULT_SS
+#define SS1                     8
+#define SS2                     9
+#define SS3                     10
+#define MOSI                    6
+#define MISO                    5
+#define SCK                     4
```
Comment (fprwi6labs) [inline variants/MAPLEMINI_F103CB/variant.h]: SPI configuration should be updated.

**Comment by Testato** [comment]
@cparata 
this is the issue we talked about at the Maker Faire Rome 2017.

The official merge of Bluepill (and all the classic board with F103C8 / CB) is of vital importance given the commercial success of these boards, all the users of Roger's core have a Bluepill / Maplemini at home
The official merge of Bluepill (and all the classic board with F103C8 / CB) is of vital importance given the commercial success of these boards, all the users of Roger's core have a Bluepill / Maplemini at home.

Official support for Bluepill has been mentioned for some time in the core homepage, but never merged.
https://github.com/stm32duino/Arduino_Core_STM32#next-release

**Comment by lacklustrlabs** [comment]
I have successfully tested this with maple mini + STLink and I would love to see this merged. 
If there is a problem with USB, couldn't that be added/fixed at a later stage?

**Comment by fpistm** [comment]
I'm currently thinking to add BP and MM without USB support.
And as you mentioned review USB later so only STLink upload and Serial via a FTDI will be available.

**Comment by fpistm** [comment]
I will provide a version end of the week.

**Comment by lacklustrlabs** [comment]
Feel free to copy&paste from my temporary [BP&MM rebase branch](https://github.com/lacklustrlabs/Arduino_Core_STM32/tree/STM32F1_merge_Other_rebased). I think I was too heavy handed with the USB stuff so I won't make a PR with it.

**Pull request closed without merging**